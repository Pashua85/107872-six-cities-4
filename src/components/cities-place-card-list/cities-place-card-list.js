import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import PlaceCardList from '../place-card-list/place-card-list';
import {getOffers} from '../../store/reducers/offersReducer/selectors';

const CitiesPlaceCardList = (props) => {
  return (
    <PlaceCardList className="cities__places-list" {...props} />
  );
};

CitiesPlaceCardList.propTypes = {
  places: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        propertyName: PropTypes.string.isRequired,
        propertyType: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]),
        propertyText: PropTypes.arrayOf(PropTypes.string),
        price: PropTypes.number.isRequired,
        rating: PropTypes.number.isRequired,
        isPremium: PropTypes.bool.isRequired,
        bedroomsAmount: PropTypes.number.isRequired,
        guestMax: PropTypes.number.isRequired,
        propertyItems: PropTypes.arrayOf(PropTypes.string),
        host: PropTypes.shape({
          name: PropTypes.string,
          avatar: PropTypes.string,
          isSuper: PropTypes.bool
        }).isRequired,
        titlePhoto: PropTypes.string.isRequired,
        photos: PropTypes.arrayOf(
            PropTypes.shape({
              title: PropTypes.string,
              id: PropTypes.string
            })
        ),
        reviews: PropTypes.arrayOf(
            PropTypes.shape({
              id: PropTypes.string,
              userName: PropTypes.string,
              avatar: PropTypes.string,
              rating: PropTypes.number,
              text: PropTypes.string
            })
        ),
        coords: PropTypes.arrayOf(PropTypes.number).isRequired
      })
  )
};

const mapStateToProps = (state) => ({
  places: getOffers(state)
});

export default connect(mapStateToProps)(CitiesPlaceCardList);
export {CitiesPlaceCardList};
