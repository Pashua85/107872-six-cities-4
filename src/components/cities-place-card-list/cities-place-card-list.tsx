import React from 'react';
import {connect} from 'react-redux';
import PlaceCardList from '../place-card-list/place-card-list';
import {getSortedCityOffers} from '../../store/reducers/offersReducer/selectors';
import {getCurrentCity} from '../../store/reducers/current-city-reducer/selectors';
import {IPlace, ICity, IStore} from '../../types/types';

interface CititesPlaceCardListProps {
  places: IPlace[],
  city: ICity
}

const CitiesPlaceCardList: React.FC<CititesPlaceCardListProps> = (props) => {
  return (
    <PlaceCardList className="cities__places-list" {...props} />
  );
};

const mapStateToProps = (state: IStore) => ({
  city: getCurrentCity(state),
  places: getSortedCityOffers(state)
});

export default connect(mapStateToProps)(CitiesPlaceCardList);
export {CitiesPlaceCardList};
