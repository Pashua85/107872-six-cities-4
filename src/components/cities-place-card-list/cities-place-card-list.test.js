import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {CitiesPlaceCardList} from './cities-place-card-list';
import places from '../../mock/test-offers';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`CitiesPlaceCardList`, () => {
  it(`should render correctly`, () => {
    const wrapper = shallow(
        <CitiesPlaceCardList places={places} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
