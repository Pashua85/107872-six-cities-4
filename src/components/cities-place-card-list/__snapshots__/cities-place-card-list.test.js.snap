// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesPlaceCardList should render correctly 1`] = `ShallowWrapper {}`;
